amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/Scheduler.java

```diff
@@ -468,19 +513,69 @@ boolean shouldSyncStreamsNow() {
         return isMultiStreamMode && (streamSyncWatch.elapsed(TimeUnit.MILLISECONDS) > NEW_STREAM_CHECK_INTERVAL_MILLIS);
     }
 
-    private void syncStreamsFromLeaseTableOnAppInit()
+    private void syncStreamsFromLeaseTableOnAppInit(List<MultiStreamLease> leases) {
+        final Set<StreamIdentifier> streamIdentifiers = leases.stream()
+                .map(lease -> StreamIdentifier.multiStreamInstance(lease.streamIdentifier()))
+                .collect(Collectors.toSet());
+        for (StreamIdentifier streamIdentifier : streamIdentifiers) {
+            if (!currentStreamConfigMap.containsKey(streamIdentifier)) {
+                currentStreamConfigMap.put(streamIdentifier, getDefaultStreamConfig(streamIdentifier));
+            }
+        }
+    }
+
+    private List<MultiStreamLease> fetchMultiStreamLeases()
             throws DependencyException, ProvisionedThroughputException, InvalidStateException {
-        if (!leasesSyncedOnAppInit && isMultiStreamMode) {
-            final Set<StreamIdentifier> streamIdentifiers = leaseCoordinator.leaseRefresher().listLeases().stream()
-                    .map(lease -> StreamIdentifier.multiStreamInstance(((MultiStreamLease) lease).streamIdentifier()))
-                    .collect(Collectors.toSet());
-            for (StreamIdentifier streamIdentifier : streamIdentifiers) {
-                if (!currentStreamConfigMap.containsKey(streamIdentifier)) {
-                    currentStreamConfigMap.put(streamIdentifier, getDefaultStreamConfig(streamIdentifier));
+        return (List<MultiStreamLease>) ((List) leaseCoordinator.leaseRefresher().listLeases());
+    }
+
+    private void removeActiveStreamsFromStaleStreamsList(Set<StreamIdentifier> streamIdentifiers) {
+        for(StreamIdentifier streamIdentifier : streamIdentifiers) {
+            staleStreamDeletionMap.remove(streamIdentifier);
+        }
+    }
+
+    private Set<StreamIdentifier> deleteMultiStreamLeases(Set<StreamIdentifier> streamIdentifiers)
```

Comment:
Let's add metrics here. This will be important to investigating any customer issues to understand when the KCL is deciding to delete streams from the lease table.

Reply:
Adding
```
private static final String ACTIVE_STREAMS_COUNT = "ActiveStreams.Count";
private static final String PENDING_STREAMS_DELETION_COUNT = "StreamsPendingDeletion.Count";
private static final String DELETED_STREAMS_COUNT = "DeletedStreams.Count";
```
under
`private static final String MULTI_STREAM_TRACKER = "MultiStreamTracker";`